### DISCUSSION TITLE
ScriptReference/ParticleSystem.SetParticles

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#534] [Notes]
# ScriptReference/ParticleSystem.SetParticles

https://docs.unity3d.com/ScriptReference/ParticleSystem.SetParticles.html

**Comment by joethephish** [comment]
If you're creating particles from scratch, remember to set all the values on each `Particle` before passing them in! In particular, you need both `startLifetime` and `remainingLifetime`, otherwise Unity will delete them immediately since it thinks they have zero life left.
